src/request.rs

```diff
@@ -38,7 +38,13 @@ where
         let original_uri = OriginalUri::from_request_parts(parts, state)
             .await
             .unwrap_or_else(|e| match e {});
-        let url = original_uri.0.path().to_string();
+        let query_string = parts
+            .uri
+            .to_string()
```

Comment:
i think we can use the [`query` method](https://docs.rs/http/latest/http/uri/struct.Uri.html#method.query) here on the `uri` to simplify this